alibi_detect/cd/chisquare.py

```diff
@@ -64,16 +64,21 @@ def __init__(
             input_shape=input_shape,
             data_type=data_type
         )
-        if categories_per_feature is None:  # infer number of possible categories for each feature from reference data
+        # construct categories from the user-specified dict
+        if isinstance(categories_per_feature, dict):
+            vals = list(categories_per_feature.values())
+            if all(isinstance(v, (int, np.int16, np.int32, np.int64)) for v in vals):
+                # categories_per_feature = Dict[int, int]
+                categories_per_feature = {f: list(np.arange(v))  # type: ignore
+                                          for f, v in categories_per_feature.items()}
+            elif not all(isinstance(v, list) for v in vals):
```

Comment:
This seems to prevent the use of `{0: None, ...}` as in the docs? I think it would be best if we have a test for each type of input `categories_per_feature` can take and check that the processing is done correctly on some dummy categorical data.

Reply:
Ah because this is `chi-square` I guess we don't have this as a valid input as all columns are categorical.

Reply:
Test would still be welcome for both detectors though...

Reply:
I will extend the current tests for both detectors.